src/main/java/com/bookmanagement/BookManagement/serviceimplementation/AuthorServiceImpl.java

```diff
@@ -0,0 +1,59 @@
+package com.bookmanagement.BookManagement.serviceimplementation;
+
+import com.bookmanagement.BookManagement.dto.AuthorDto;
+import com.bookmanagement.BookManagement.dto.BookDto;
+import com.bookmanagement.BookManagement.entity.Author;
+import com.bookmanagement.BookManagement.mapper.AuthorMapper;
+import com.bookmanagement.BookManagement.mapper.BookMapper;
+import com.bookmanagement.BookManagement.repository.AuthorRepository;
+import com.bookmanagement.BookManagement.repository.BookRepository;
+import com.bookmanagement.BookManagement.service.AuthorService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service("authorService")
+public class AuthorServiceImpl implements AuthorService {
+    private final AuthorRepository authorRepository;
+    private final AuthorMapper authorMapper;
+    private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
```

Comment:
Ideally, if we have 2 independent/main entities then we don't inject the repo/mapper dependencies of one entity into another's service. 
Here, inject book service and perform all the book-related operations via book service.

In the case of student and address/bag use case, this was correct as address/bag was heavily dependent on the student.

Reply:
Done